resources/authoritative_service_sysvinit.rb

```diff
@@ -25,31 +25,29 @@
 property :instance_name, String, name_property: true, callbacks: {
   'should not contain a hyphen' => ->(param) { !param.include?('-') },
 }
-property :cookbook, [String, NilClass], default: 'pdns'
-property :source, [String, NilClass]
+property :cookbook, String, default: 'pdns'
+property :source, String
 property :config_dir, String, default: lazy { default_authoritative_config_directory }
 property :variables, String
 
 action :enable do
-  if new_resource.source
-    template "/etc/init.d/#{sysvinit_name}" do
-      source new_resource.source
-      owner 'root'
-      group 'root'
-      mode '0755'
-      variables(
-        variables: new_resource.variables
-      )
-      action :create
-    end
-  else
-    # Has specified in the PowerDNS documentation, a symlink to the init.d script
-    # "pdns" should be enough for setting up a Virtual instance:
-    # https://github.com/PowerDNS/pdns/blob/master/docs/markdown/authoritative/running.md#starting-virtual-instances-with-sysv-init-scripts
-    link "/etc/init.d/#{sysvinit_name(new_resource.instance_name)}" do
-      to 'pdns'
-      not_if { new_resource.instance_name.empty? }
-    end
+  template "/etc/init.d/#{sysvinit_name}" do
+    source new_resource.source
+    owner 'root'
+    group 'root'
+    mode '0755'
+    variables(
+      variables: new_resource.variables
+    )
+    action :create
+    only_if { new_resource.property_is_set?(:source) }
+  end
+
+  # Create a symlink to the original pdns script to make a virtual instance
+  # https://doc.powerdns.com/md/authoritative/running/#virtual-hosting
+  link "/etc/init.d/#{sysvinit_name(new_resource.instance_name)}" do
+    to '/etc/init.d/pdns'
+    not_if { new_resource.instance_name.empty? || new_resource.property_is_set?(:source) }
```

Comment:
Basically what we want to do for the authoritative is defer to their packaged init versus attempting to override it by default. If you actually do override it, you're signing on the dotted line that you know what you are doing and we stop attempting to symlink anything. There is something to be said for supporting the symlinking, but I'd want someone to propose a patch with a use case then.